File: Caesar-Ciphers/cc.js
//Gregory Ecklund
//September 2022

let text = "You can make this say whatever you want."; //Text to encrypt/decrypt
let rotationInteger = 1; //Amount to rotate cipher by (positive or negative)

let lowercaseAlphabet = "abcdefghijklmnopqrstuvwxyz";
let uppercaseAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
let remainingCharacters = "1234567890 ?,\'\"./-_+=][}{;:><~`\\|!@#$%^&*()";

output = "";
let newIndex = 0;
for (let letter of text) {
    for (let i = 0; i < lowercaseAlphabet.length; i++) {
        if (letter === lowercaseAlphabet[i]) {
            newIndex = i + rotationInteger;
            if (newIndex > 25) {
                newIndex %= 26;
            }
            output += lowercaseAlphabet[newIndex];
        }
    }
    for (let j = 0; j < uppercaseAlphabet.length; j++) {
        if (letter === uppercaseAlphabet[j]) {
            newIndex = j + rotationInteger;
            if (newIndex > 25) {
                newIndex %= 26;
            }
            output += uppercaseAlphabet[newIndex];
        }
    }
    for (let k of remainingCharacters) {
        if (letter === k) {
            output += letter;
        }
    }
}

console.log(output);

File: Caesar-Ciphers/cc.py
#Gregory Ecklund
#September 2022

text = "You can make this say whatever you want." #Text to encrypt/decrypt
rotation_integer = 1 #Amount to rotate cipher by (positive or negative)

lowercase_alphabet = "abcdefghijklmnopqrstuvwxyz"
uppercase_alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
remaining_characters = "1234567890 ?,\'\"./-_+=][}{;:><~`\\|!@#$%^&*()"

output = ""
new_index = 0
for letter in text:
    for i in range(len(lowercase_alphabet)):
        if letter == lowercase_alphabet[i]:
                new_index = i + rotation_integer
                if new_index > 25:
                    new_index %= 26
                output += lowercase_alphabet[new_index]
    for j in range(len(uppercase_alphabet)):
        if letter == uppercase_alphabet[j]:
            new_index = j + rotation_integer
            if new_index > 25:
                new_index %= 26
            output += uppercase_alphabet[new_index]
    for k in remaining_characters:
        if letter == k:
            output += letter

print(output)
